src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -173,13 +182,12 @@ public GatewayMetaState(Settings settings, ThreadPool threadPool, NodeEnvironmen
                 GATEWAY_DANGLING_TIMEOUT, this.danglingTimeout);
         if (DiscoveryNode.masterNode(settings) || DiscoveryNode.dataNode(settings)) {
             nodeEnv.ensureAtomicMoveSupported();
-        }
-        if (DiscoveryNode.masterNode(settings)) {
             try {
                 ensureNoPre019State();
                 pre20Upgrade();
                 long start = System.currentTimeMillis();
-                loadState();
+                MetaData metaData = loadState();
+                ensureAllIndicesAreSupported(metaData);
```

Comment:
I wonder if we should have an escape hatch (some property etc) to disable the check. I've seen index corruptions manifest themselves as too-old/too-new exceptions before.  I'm worried about a user being "stuck" where elasticsearch won't start, but maybe there is some other way they could move past this?